libnetwork/network.go

```diff
@@ -1302,7 +1302,12 @@ func (n *Network) EndpointByID(id string) (*Endpoint, error) {
 }
 
 // updateSvcRecord adds or deletes local DNS records for a given Endpoint.
-func (n *Network) updateSvcRecord(ep *Endpoint, isAdd bool) {
+func (n *Network) updateSvcRecord(ctx context.Context, ep *Endpoint, isAdd bool) {
```

Comment:
Now that this function takes a context argument and it would be reasonable for it to become cancelable in the future, it should return an error so that callers can find out if the call was canceled.
```suggestion
func (n *Network) updateSvcRecord(ctx context.Context, ep *Endpoint, isAdd bool) error {
```

Reply:
It might end up becoming cancelable at a latter time, but I don't think it's worthwile to make it return a nil error for the sake of that. So, I prefer to keep it as-is for now, and just make sure every callsite pass a non-cancelable context.

Reply:
Accepting a context and not retuning an error signifies that the function will never fail, even if the context is done. For example: `log.G()`. Having the function return an error acts as a reminder to the caller that they need to take context cancelation into consideration when calling it. (When I review context plumbing code, I'm scanning for `if err := fn(ctx, ...); err != nil {` and related patterns to flag potential cancelation issues. They're always there because the linter enforces explicit checking or discarding of error return values.) Without the error return, it is that much more error prone to "just make sure."

Reply:
My premise is wrong: it would _not_ be reasonable for this function to be cancelable in the future. My bad.